### PR TITLE
feat: include build version in Slack report header

### LINKED ISSUE #194: feat: include build version in Slack report header
## Problem

The Slack report header currently shows:

```
oompa report -- 3 project(s) with activity
```

There's no way to tell which oompa version generated the report. After a version update restart, it's useful to confirm the new version is running and reporting.

## Fix

Include the build version (short commit SHA) in the Slack report header:

```
oompa report (38767d1) -- 3 project(s) with activity
```

The commit SHA is already available via `debug.ReadBuildInfo()` VCS revision (used by `exit-on-new-version`). Pass it to the `SlackReporter` and include it in the header format string.

## Files to change

| File | Change |
|------|--------|
| `pkg/agent/slack.go` | Add `Version` field to `SlackReporter`, include in header format string at line 263 |
| `cmd/oompa/main.go` | Pass `commitSHA` (or short version) when creating the `SlackReporter` |

### PR DESCRIPTION
**Pull request by qinqon**
Fixes #194

## Summary

Include the build version (short commit SHA) in the Slack report header so operators can confirm which oompa build generated a report, especially useful after version updates.

## Changes

- Add `version` field to `SlackReporter` struct and `NewSlackReporter` constructor
- Update `formatSlackMessage` to accept a version parameter and include it in the header when non-empty
- Pass `cfg.Version` (commit SHA) when creating `SlackReporter` in both single-repo and multi-project modes
- Add tests for version inclusion and empty-version omission in the header
- Update `specs/slack.md` to reflect the new signatures and header format

## Testing

- [x] `go test ./...` passes
- [x] `go vet ./...` passes
- [x] New tests verify version appears in header as short SHA
- [x] New tests verify empty version produces original header format (no empty parens)

## Related Issues

Fixes #194

## Summary by CodeRabbit

* **New Features**
  * Slack reports now include version information (commit SHA) in the message header, helping track which build generated each report.

* **Documentation**
  * Updated Slack reporting documentation to reflect version inclusion in messages.

[Review Change Stack](https://app.coderabbit.ai/change-stack/qinqon/oompa/pull/197?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by qinqon** [comment]
## CI Failure Analysis — commit 7cdd70f

**Total failures**: 1 (1 related)
**Action taken**: Pushed fix for related failure

---

<details>
<summary>🔴 Related: <code>test</code> — fix pushed</summary>

### Error
````
```
pkg/agent/slack.go:68:1: paramTypeCombine: func(webhookURL string, version string, logger *slog.Logger) *SlackReporter could be replaced with func(webhookURL, version string, logger *slog.Logger) *SlackReporter (gocritic)
func NewSlackReporter(webhookURL string, version string, logger *slog.Logger) *SlackReporter {
```
````

### Root Cause
The `NewSlackReporter` function signature used `webhookURL string, version string` with the type repeated for consecutive parameters of the same type. The `gocritic` linter's `paramTypeCombine` rule requires combining them as `webhookURL, version string`. This was introduced by this PR when adding the `version` parameter.

### Recommendation
Fixed by combining the parameter types: `func NewSlackReporter(webhookURL, version string, logger *slog.Logger)`. Verified with `golangci-lint run ./...` (0 issues) and `go test ./...` (all pass).

### Action
Pushed fix for this failure.

</details>

Pushed a fix for the related failure.

File: pkg/agent/slack.go [hunk 1]
```diff
@@ -201,19 +204,20 @@ func projectKey(owner, repo string) string {
...
 
 // formatSlackMessage builds a Slack Block Kit JSON payload grouped by project and then by PR.
-// Returns nil if there are no findings.
+// Returns nil if there are no findings. The version string (short commit SHA) is included
+// in the header when non-empty.
 //
 // Format:
 //
-//	🏭 oompa report — N projects with activity
+//	🏭 oompa report (38767d1) — N projects with activity
```
[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/agent/slack.go]: > The format example in this comment doesn't match the actual header string produced below: the code emits "project(s)" but the comment says "projects". Please update the example to match the real output so future readers don't get misled.

Addressed: Updated the doc comment on line 212 from `N projects with activity` to `N project(s) with activity` to match the actual format string.
